src/libraries/System.Private.CoreLib/src/System/IO/BufferedStream.cs

```diff
@@ -609,13 +629,13 @@ public override Task<int> ReadAsync(byte[] buffer, int offset, int count, Cancel
             // an Async operation.
             SemaphoreSlim sem = EnsureAsyncActiveSemaphoreInitialized();
             Task semaphoreLockTask = sem.WaitAsync(cancellationToken);
-            if (semaphoreLockTask.IsCompletedSuccessfully)
+            bool locked = semaphoreLockTask.IsCompletedSuccessfully;
```

Comment:
Nit: `locked` is a bit ambiguous... did we acquire the lock?  Is it already "locked" and thus we couldn't acquire it?  How about `acquiredLock` or `gotLock` or something like that?